### PR TITLE
Change default for LAMBDA_DEV_PORT_EXPOSE on macOS in host mode

### PR DESCRIPTION
**Pull request by joe4dev**
## Motivation

Lambda won't work in host mode on macOS without setting `LAMBDA_DEV_PORT_EXPOSE=1` (introduced in https://github.com/localstack/localstack/pull/7367), which is a bad developer experience.
So far, we opted for explicit configuration developer documentation [here](https://github.com/localstack/localstack/blob/master/docs/development-environment-setup/README.md#lambda) to unblock developers on macOS in host mode.
Several contributors hit this issue, and it was suggested that we set it by default if we detect LocalStack running in host mode on macOS. The Lambda service owners also discussed this in the last Lambda Service Owner Sync (2024-07-15).

## Changes

* [Unify helper utils for common operating systems](https://github.com/localstack/localstack/commit/d087e5bd64a9b728173053c10c845deadff78a8e)
* [Change default for Lambda port expose on macOS in host mode](https://github.com/localstack/localstack/commit/ae3a0c52eba2ba7ff3d4f45d0e3736eecdb2bbaf)

## Testing

Our test suite should capture any regressions. Anything else to consider? Is there a special environment where the assumption `not is_in_docker == host mode` does not apply (e.g., K8) @dfangl @simonrw?

## Discussion

Options considered:

- Option A: simply evaluate based on is_in_docker and the system platform, assuming `not is_in_docker == host mode`.
    - Mixing up config logic with production code makes config bugs more likely and leads to more complex config code
- Option B: introduce some explicit host mode LS devs should set for sensible default config
    - Could separate special host mode defaults from generic production defaults

### CODE REVIEW

**Comment by gregfurman** [comment]
Nice change! Tested locally running on host mode (without explicitly enabling the flag) on `master` and the `lambda-defaults-for-macos-hostmode` feature branch and the dev ports expose successfuly by default with these changes.

💻  I'm running `Darwin 23.3.0 arm64`

**Review by dfangl — APPROVED** [review]
is_in_docker will also be `True` if in kubernetes, it is really just a variable whether we are running in a container or not.

I think the taken approach is fine, but there are now some duplicated methods with utils/platform.py. I understand it is not that easy to use them, but it might be worth consolidating them anyway in the future.

**Comment by joe4dev** [comment]
> is_in_docker will also be `True` if in kubernetes, it is really just a variable whether we are running in a container or not.
> 
> I think the taken approach is fine, but there are now some duplicated methods with utils/platform.py. I understand it is not that easy to use them, but it might be worth consolidating them anyway in the future.

Yeah, I know, and also small inconsistencies (e.g., `lower()`, `is_mac_os()`). I thought that was intentional because config should be self-contained for bootstrapping 🤷

**Review by simonrw — COMMENTED** [review]
I think this change is sensible.

Regarding k8s behaviour: I think the same problem will be true for k8s as it is in Docker, but only when LS is being run in host mode. E.g. for `kind` the k8s "nodes" are Docker containers, therefore the same problem.

However we implicitly not really supporting that use case right now. We are focused on when LS is deployed in the k8s cluster as a pod, in which case there is no problem as all pods ports are accessible by other pods (with minor restrictions).

So in summary: I think we don't have to account for k8s behaviour specifically here.
